chall04/pcuadrad.py

```diff
@@ -0,0 +1,80 @@
+#!/usr/bin/env python3
+
+import sys
+
+def error_message(filename, message):
+    print (sys.argv[0] + ": " + filename + ": " + message)
+
+def place_books(books, shelfs):
+    rest = 0
+    last_book = 0
+    for i in range(0, len(shelfs)):
+        begin = shelfs[i]
+        begin += rest
+        rest = 0
+        if begin < 0 and i is len(shelfs) - 1:
+            return False
+        for j in range(last_book, len(books)):
+            begin -= books[j]
+            if begin < 0:
+                rest = begin
+                last_book = j + 1
+                break
+            if j is len(books) - 1:
+                print (i + 1)
+                return True
+    print (i + 1)
+    return True
+
+def parse_file(txt):
+    i = 0
+    books = []
+    shelfs_int = []
+    for line in txt:
+        if i is 0:
+            shelfs = line.split(' ')
+        else:
+            line2 = line.split(' ', 1)
+            books.append(int(line2[0]))
+        i += 1
+    [shelfs_int.append(int(i)) for i in shelfs]
+    shelfs_int = sorted(shelfs_int, reverse=True)
```

Comment:
You didn't quite understand the user of comprehension, a comprehension will run and generate a list directly on memory, meaning that if you don't assign it, it is lost.
Here is a little [documentation about comprehensions](https://www.pythonforbeginners.com/basics/list-comprehensions-in-python). I could not explain it better.
```suggestion
    shelfs_int = sorted([int(i) for i in shelfs], reverse=True)
```